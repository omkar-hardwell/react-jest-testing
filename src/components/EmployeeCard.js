import React from "react";
import { Button, Card, Image } from "semantic-ui-react";

import emppic from "../images/steve.jpg";

const EmployeeCard = ({ ...props }) => {
  return (
    <Card>
      <Card.Content>
        <Image floated="right" size="mini" src={emppic} />
        <Card.Header>{props.employee.name}</Card.Header>
        <Card.Meta>Emp ID - {props.employee.id}</Card.Meta>
        <Card.Meta>Designation - {props.employee.designation}</Card.Meta>
        <Card.Meta>Department - {props.employee.department}</Card.Meta>
      </Card.Content>
      <Card.Content extra>
        <div className="ui two buttons">
          <Button basic color="green">
            Edit
          </Button>
          <Button basic color="red">
            Delete
          </Button>
        </div>
      </Card.Content>
    </Card>
  );
};

export default EmployeeCard;
