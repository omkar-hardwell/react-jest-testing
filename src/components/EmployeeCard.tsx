import React from "react";
import { Button, Card, Image } from "semantic-ui-react";

import emppic from "../images/steve.jpg";
import { IEmployee } from "./models";

interface IEmployeeInfo {
  employee: IEmployee;
}

const EmployeeCard: React.FC<IEmployeeInfo> = ({ employee }) => {
  return (
    <Card>
      <Card.Content>
        <Image floated="right" size="mini" src={emppic} />
        <Card.Header>{employee.name}</Card.Header>
        <Card.Meta>Emp ID - {employee.id}</Card.Meta>
        <Card.Meta>Designation - {employee.designation}</Card.Meta>
        <Card.Meta>Department - {employee.department}</Card.Meta>
      </Card.Content>
      <Card.Content extra>
        <div className="ui two buttons">
          <Button basic color="green">
            Edit
          </Button>
          <Button basic color="red">
            Delete
          </Button>
        </div>
      </Card.Content>
    </Card>
  );
};

export default EmployeeCard;
