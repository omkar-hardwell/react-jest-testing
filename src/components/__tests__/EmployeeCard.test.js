import React from "react";
import { shallowToJson } from "enzyme-to-json";
import { shallow } from "enzyme";

import EmployeeCard from "../EmployeeCard";

import "../../setupTests.js";

describe("EmployeeCard testing with", () => {
  let wrapper;

  const dummyEmployeeInfo = {
    id: 1,
    name: "Omkar",
    designation: "React developer",
    department: "UI",
  };
  it("render card", () => {
    wrapper = shallow(<EmployeeCard employee={dummyEmployeeInfo} />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
