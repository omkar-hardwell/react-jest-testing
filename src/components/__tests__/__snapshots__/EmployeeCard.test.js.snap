// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeCard testing with render card 1`] = `
<Card>
  <CardContent>
    <Image
      as="img"
      floated="right"
      size="mini"
      src="steve.jpg"
      ui={true}
    />
    <CardHeader>
      Omkar
    </CardHeader>
    <CardMeta>
      Emp ID - 
      1
    </CardMeta>
    <CardMeta>
      Designation - 
      React developer
    </CardMeta>
    <CardMeta>
      Department - 
      UI
    </CardMeta>
  </CardContent>
  <CardContent
    extra={true}
  >
    <div
      className="ui two buttons"
    >
      <Button
        as="button"
        basic={true}
        color="green"
      >
        Edit
      </Button>
      <Button
        as="button"
        basic={true}
        color="red"
      >
        Delete
      </Button>
    </div>
  </CardContent>
</Card>
`;
